### DISCUSSION TITLE
Getting an java.lang.IllegalStateException on message.delete().queue()

### DISCUSSION THREAD

**Discussion by Beutilo** [discord-jda/JDA#1574] [Questions and Help]
Hey guys!
Following problem.

I have a bot which answers in the private channel (dm). I have to "delete" the messages of the users of the queue in order to not have duplicate answers.
For example if I do not delete the message)

User msg: !info
Bot: xanswer
User msg: !info
Bot: xanswer
Bot: xanswer

Now I fixed the problem by the following quote, the problem is that I get an exception in the console (it still works tho).

```java
if (event.isFromType(ChannelType.PRIVATE))
		{
			if(msg.startsWith(DiscordBotHauptklasse.prefix))
			{
				switch (msg)
				{
					case "!info":
						message.getChannel().sendMessage(Messages.info).queue();
						message.delete().queue();
						break;
					case "!infoextra":
						message.getChannel().sendMessage(Messages.infoextra).queue();
						message.delete().queue();
						break;
					default:
						message.getChannel().sendMessage("Ungültiger Command.").queue();
						message.delete().queue();
						break;
				}
```

Here a picture:

![grafik](https://user-images.githubusercontent.com/55931345/113354737-e76f8400-933f-11eb-827f-f735fc4b7ede.png)

**Comment by TheDGOfficial** [comment]
The stacktrace and error does not match. The stacktrace says you’ve called
getGuild in line 21 but you have getMessage there. Are you showing the
correct code?

Neverthless, the error is you’ve tried to get a Guild object when the
message is not from one; and you should get help through Discord, not here.

**Reply by Beutilo** [reply]
![grafik](https://user-images.githubusercontent.com/55931345/113393331-cd14c500-9396-11eb-88e3-cc82f003e687.png)

![grafik](https://user-images.githubusercontent.com/55931345/113393361-da31b400-9396-11eb-804a-d79b260a22a0.png)


Here the new picture

Which Discord?

**Reply by gpluscb** [reply]
For this issue it is exactly like the exception suggests: You can't delete another users messages in DMs.
By the way, the call to `super.onMessageReceived` is not necessary, it does nothing.
You can join the JDA Discord server DG mentioned over here -> https://discord.gg/0hMr4ce0tIk3pSjp

**Reply by Beutilo** [reply]
Alright. 
Thanks

**Reply by DV8FromTheWorld** [reply]
To be clear, while real-time help is providable in the Discord (and historically been recommended over Github Issues), the reason we opened up Github Discussions is to facilitate stack overflow like assistance.

**Comment by Andre601** [comment]
The best way to fix this issue, if you still want to delete messages when possible is to either use `GuildMessageReceivedEvent` if your system should only work within a guild, or do some checks for if the message is from a guild and if your bot has the right permissions.

(Also, don't forget to always select an response as solution if it helped you solving the issue, as that helps others finding solutions to similar problems.)
